docs/design-documents/nfc/nfc_design.md

```diff
@@ -0,0 +1,661 @@
+```
+
+Retrieve the list of supported RF protocols. These are mapped against NFC Forum-defined protocols.
+
+- T1T is based on ISO/IEC 14443A-3 and commonly known as Topaz (Innovision).
+- T2T is based on ISO/IEC 14443A-3 and commonly known as Mifare Ultralight/NTAG (NXP).
+- T3T is based on JIS X6319-4, also known as Felica (Sony).
+- ISO-DEP is based on ISO/IEC 14443-4 and is the common interface for contactless smartcards. The underlying radio protocol can either be ISO/IEC 14443A or ISO/IEC 14443B.
+- NFC-DEP is based on ISO/IEC 18092/FIXME and is the basis for NFC peer-to-peer communication.
+- T5T is also known as ISO/IEC 15963.
+
+```cpp
+nfc_err_t initialize();
+```
+
+Initialize the NFC controller.
+
+```cpp
+nfc_err_t configure_rf_protocols(nfc_rf_protocols_bitmask_t rf_protocols);
+```
+
+Configure which protocols should be enabled during the discovery process.
+
+```cpp
+nfc_err_t start_discovery();
+```
+
+Start the discovery process.
+
+```cpp
+nfc_err_t cancel_discovery();
+```
+
+Cancel the discovery process (if running).
+
+**Delegate**
+
+A `NFCController` instance needs to be configured with a delegate to receive events.
+
+```cpp
+enum nfc_discovery_terminated_reason_t {
+    nfc_discovery_terminated_completed = 0
+    nfc_discovery_terminated_canceled,
+    nfc_discovery_terminated_rf_error
+};
+
+void on_discovery_terminated(nfc_discovery_terminated_reason_t reason);
+```
+
+Let the user know when a discovery loop has been terminated (either because endpoints have been found, the user canceled it or an error occurred).
+
+```cpp
+void on_nfc_initiator_discovered(const mbed::SharedPtr<NFCRemoteInitiator>& nfc_initiator);
+
+void on_nfc_target_discovered(const mbed::SharedPtr<NFCRemoteTarget>& nfc_target);
+```
+
+These methods called when a remote initiator (the local controller is acting as a target) or a remote target (the local controller is acting as an initiator) is detected.
+
+Shared pointers are used, so the user does not have to maintain the lifetime of these objects. The `NFCController` instance releases its reference when the endpoint is lost (see below).
+
+### Endpoints
+
+![nfc_endpoints_diagram]
+
+#### NFC remote endpoint
+
+A remote endpoint is a generic NFC-enabled device with which the controller is communicating over the air interface:
+
+```cpp
+nfc_err_t connect();
+```
+
+Establish a connection with the remote endpoint.
+
+```cpp
+nfc_err_t disconnect();
+```
+
+Drop the connection with the remote endpoint.
+
+```cpp
+bool is_connected() const;
+```
+
+Set to true when the connection to this endpoint has been activated and is currently selected by the controller.
+
+```cpp
+bool is_disconnected() const;
+```
+
+Set to true when this endpoint has been lost and the controlled instance has released the reference to the shared pointer.
+
+```cpp
+nfc_rf_protocols_bitmask_t rf_protocols() const;
+```
+
+List the RF protocols that have been activated to communicate with that endpoint.
```

Comment:
This document uses a lot of passive voice [have been activated, is called, will be used and so on]. Please rewrite all of the passive voice in active voice, so the reader knows who or what is doing the action. What has activated the protocols? What calls the function? Who or what will use the buffer?